File: week-2/question_23.js
function factorial(num){
    /**
     * returns the factorial of a number
     * @param {Integer} num - the given input 
     * @return {Integer} - returns the factorial of the number
     * @throws {TypeError} - if the input is not a positive integer
     * @throws {Error} - if the integer is not positive
     */
    if(!Number.isInteger(num)){
        throw new TypeError("Please input a postive integer")
    }if(num < 0){
        throw new Error("Please enter a positive integer")
    }
    if(num == 0 ){
        return 1
    }
    return num * factorial(num-1)
}
module.exports = factorial;

File: week-1/question_23.py
# fibonacci series up to N terms
series = []

def fibonacci(num):
    """
    returns the Fibonacci sequence up to num terms

    Args:
        num (int): starting point
    
    Returns:
        str: a sequence of the fibonacci series up to the num terms

    Raises:
        ValueError: if num is not integer, or not postive
    """
    try:
        num = int(num)
    except ValueError:
        raise ValueError("Please Enter a Valid integer")
    if num < 0:
        raise ValueError("Please Enter a postive Integer")

    i = 0
    while i < num:
        if i == 0 or i == 1: 
            series.append(i) 
            i += 1 
        else:
            value = series[-1] + series[-2]
            series.append(value)
            i += 1 
    return series
    
    